
message Tank {
	required int32 id = 1;
	required int32 team = 2;
	required sint32 posx = 3;
	required sint32 posy = 4;
}

message FieldStatus {
	repeated Tank tanks = 1;
}

message LoginRequest {
	required sint32 team = 1;
	required string password = 2;
}

message LoginResponse {
	required string session = 1;
}

message LogoutRequest {
	required string session = 1;
}

message LogoutResponse {
	required bool success = 1;
}

message StatusRequest {
	required string session = 1;
}

message StatusResponse {
	required bool success = 1;
	optional FieldStatus fieldStatus = 2;
	optional int32 turnNum = 3;
}

message WaitForSimulationRequest {
	required string session = 1;
}
message WaitForSimulationResponse {
	required int32 turnNum = 1;
}

service CodebotsService {
	rpc login(LoginRequest) returns(LoginResponse);
	rpc logout(LogoutRequest) returns(LogoutResponse);
	rpc getStatus(StatusRequest) returns(StatusResponse);
	rpc waitForSimulation(WaitForSimulationRequest) returns(WaitForSimulationResponse);
}

